### PR TITLE
[NCL-7943] Use the pnc-api Request obj for callback

### PR DESCRIPTION
**Pull request by thescouser89**

### CODE REVIEW

**Comment by thescouser89** [comment]
@michalovjan I do worry about the confusion that this will create between the Request object from pnc-api and the Request object from rex.

File: core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java [hunk 1]
```diff
@@ -65,16 +65,26 @@ public void stopJob(Task task) {
...
         } catch (URISyntaxException e) {
             throw new IllegalArgumentException("remoteCancel.url is not a valid URL for task with name " +
                     task.getName(), e);
         }
 
+        String callback = baseUrl + "/rest/internal/"+ task.getName() + "/finish";
+
+        try {
+            URI callbackUri = new URI(callback);
+            callbackRequest = new org.jboss.pnc.api.dto.Request(org.jboss.pnc.api.dto.Request.Method.POST, callbackUri);
```
Comment (michalovjan) [inline core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java:80]: can you extract this into method? and also use it in startJob()

Comment (michalovjan) [inline core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java:80]: The service that calls back will have to add a Content-type

File: model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java [hunk 2]
```diff
@@ -31,7 +32,7 @@
 @ToString
 public class StartRequest {
 
-    private final String callback;
+    private final Request callback;
```
Comment (michalovjan) [inline model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java:35]: can you specify the full java package here to address the probable confusion between Request classes

File: model/src/main/java/org/jboss/pnc/rex/model/requests/StopRequest.java [hunk 3]
```diff
@@ -31,7 +32,7 @@
 @ToString
 public class StopRequest {
 
-    private final String callback;
+    private final Request callback;
```
Comment (michalovjan) [inline model/src/main/java/org/jboss/pnc/rex/model/requests/StopRequest.java:35]: can you specify the full java package here to address the probable confusion between Request classes

**Review by michalovjan — CHANGES_REQUESTED** [review]

**Comment by michalovjan** [comment]
> @michalovjan I do worry about the confusion that this will create between the Request object from pnc-api and the Request object from rex.

agreed, let's specify the full package inside the StartRequest and StopRequest, that should alert the reader of those classes

File: model/src/main/java/org/jboss/pnc/rex/model/requests/StopRequest.java (hunk 3 above)
Comment (thescouser89) [reply model/src/main/java/org/jboss/pnc/rex/model/requests/StopRequest.java:35]: Yup will do!

File: core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java (hunk 1 above)
Comment (thescouser89) [reply core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java:80]: yup done!

Comment (thescouser89) [reply core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java:80]: Yup!

**Comment by thescouser89** [comment]
@michalovjan Pushed the changes! Thanks for the suggestions :)

I would also like to note that I am not deliberately   passing any Authorization headers since it's still a bit unclear how we'll handle them. Let me know what you think! :)

**Review by michalovjan — APPROVED** [review]

**Pull request merged**